deepspeed/runtime/pipe/module.py

```diff
@@ -342,7 +365,29 @@ def exec_func(*inputs):
                     inputs = layer(inputs)
                 return inputs
 
-            return exec_func
+            def exec_func_warp(*inputs):
```

Comment:
Can we consolidate this with exec_func?
Most of the code are duplicated.

Reply:
merged.